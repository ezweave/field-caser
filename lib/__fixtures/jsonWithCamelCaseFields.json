{
	"firstName": "Joe",
	"lastName": "Consumer",
	"details": {
		"dateOfBirth": "1/1/1970",
		"cars": [{
			"brand": "Pontiac",
			"model": "Aztek",
			"yearOfManufacture": "2004"
		},{
			"brand": "Saturn",
			"model": "Vue",
			"yearOfManufacture": "2004"
		}]
	}
}
