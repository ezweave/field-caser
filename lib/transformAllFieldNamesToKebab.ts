import { kebabCase } from 'lodash'
import { deepFieldTransformer } from './deepFieldTransformer'

export const transformAllFieldNamesToKebab: <T>(source: unknown) => T = deepFieldTransformer(kebabCase)