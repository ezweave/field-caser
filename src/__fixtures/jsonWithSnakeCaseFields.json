{
	"first_name": "Joe",
	"last_name": "Consumer",
	"details": {
		"date_of_birth": "1/1/1970",
		"cars": [{
				"brand": "Pontiac",
				"model": "Aztek",
				"year_of_manufacture": "2004"
			},{
				"brand": "Saturn",
				"model": "Vue",
				"year_of_manufacture": "2004"
		}]
	}
}
