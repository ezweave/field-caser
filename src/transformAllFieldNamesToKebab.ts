import { kebabCase } from 'lodash'
import { deepFieldTransformer } from './deepFieldTransformer'

export const transformAllFieldNamesToKebab = deepFieldTransformer(kebabCase)