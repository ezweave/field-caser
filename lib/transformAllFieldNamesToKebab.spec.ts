import { transformAllFieldNamesToKebab } from './transformAllFieldNamesToKebab'
import * as jsonWithSnakeCaseFields from './__fixtures/jsonWithSnakeCaseFields.json'
import * as jsonWithCamelCaseFields from './__fixtures/jsonWithCamelCaseFields.json'

describe('transformAllFieldNamesToKebab', () => {
  it('transforms camel field names to kebab', () => {
    expect(transformAllFieldNamesToKebab(jsonWithCamelCaseFields)).toMatchSnapshot()
  })
  it('transforms snake field names to kebab', () => {
    expect(transformAllFieldNamesToKebab(jsonWithSnakeCaseFields)).toMatchSnapshot()
  })
})