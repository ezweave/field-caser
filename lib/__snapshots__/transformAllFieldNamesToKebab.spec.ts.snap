// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformAllFieldNamesToKebab transforms camel field names to kebab 1`] = `
Object {
  "default": Object {
    "details": Object {
      "cars": Array [
        Object {
          "brand": "Pontiac",
          "model": "Aztek",
          "year-of-manufacture": "2004",
        },
        Object {
          "brand": "Saturn",
          "model": "Vue",
          "year-of-manufacture": "2004",
        },
      ],
      "date-of-birth": "1/1/1970",
    },
    "first-name": "Joe",
    "last-name": "Consumer",
  },
  "details": Object {
    "cars": Array [
      Object {
        "brand": "Pontiac",
        "model": "Aztek",
        "year-of-manufacture": "2004",
      },
      Object {
        "brand": "Saturn",
        "model": "Vue",
        "year-of-manufacture": "2004",
      },
    ],
    "date-of-birth": "1/1/1970",
  },
  "first-name": "Joe",
  "last-name": "Consumer",
}
`;

exports[`transformAllFieldNamesToKebab transforms snake field names to kebab 1`] = `
Object {
  "default": Object {
    "details": Object {
      "cars": Array [
        Object {
          "brand": "Pontiac",
          "model": "Aztek",
          "year-of-manufacture": "2004",
        },
        Object {
          "brand": "Saturn",
          "model": "Vue",
          "year-of-manufacture": "2004",
        },
      ],
      "date-of-birth": "1/1/1970",
    },
    "first-name": "Joe",
    "last-name": "Consumer",
  },
  "details": Object {
    "cars": Array [
      Object {
        "brand": "Pontiac",
        "model": "Aztek",
        "year-of-manufacture": "2004",
      },
      Object {
        "brand": "Saturn",
        "model": "Vue",
        "year-of-manufacture": "2004",
      },
    ],
    "date-of-birth": "1/1/1970",
  },
  "first-name": "Joe",
  "last-name": "Consumer",
}
`;
